frontend/pages/SoftwarePage/helpers.ts

```diff
@@ -20,8 +20,17 @@ export const generateSecretErrMsg = (err: unknown) => {
   }
 
   if (errorType === "profile") {
-    return reason
```

Comment:
How about using `.pop()` here? Gets rid of the branching logic with the same end result.

Reply:
makes sense. updated to this approach.